### ISSUE TITLE
Unable to connect to ftp server from EKS

### ISSUE THREAD

**Issue by itsthatriver** [cilium/cilium#16232]
## Bug report

**General Information**

- Cilium version: found with both 1.9.7 and 1.9.5
- Kernel version: Linux my-shell 5.4.110-54.189.amzn2.x86_64 #1 SMP Mon Apr 26 21:25:38 UTC 2021 x86_64 GNU/Linux
- Orchestration system version in use: v1.19.6-eks-49a6c0
- Link to relevant artifacts: We only have one active networkpolicy selecting the pod where these tests were run, and it is described below
Spec:
  PodSelector:     <none> (Allowing the specific traffic to all pods in this namespace)
  Allowing ingress traffic:
    To Port: <any> (traffic allowed to all ports)
    From:
      NamespaceSelector: trusted-traffic=true
  Not affecting egress traffic
  Policy Types: Ingress
- Generate and upload a system zip:
[cilium-sysdump.zip](https://github.com/cilium/cilium/files/6511797/cilium-sysdump.zip)

**How to reproduce the issue**

1. Create an EKS cluster, version 1.19.7 (we use r5.2xlarge with amazon-linux-2 images - the default eks-optimized image)
2. Use the following helm command to set it up, but it has also been confirmed as present with cluster-pool ipam, and with both generic and aws cilium-operator
3. helm install cilium cilium/cilium --version 1.9.7 \
  --namespace kube-system \
  --set eni=true \
  --set ipam.mode=eni \
  --set egressMasqueradeInterfaces=eth0 \
  --set nodeinit.enabled=true
3. exec into a pod in the cluster and run either of the following commands
4. nc -zv speedtest.tele2.net 21
5. host myip.opendns.com resolver1.opendns.com
6. They should both fail, even though they work just fine locally (and we found them to work fine in GKE with cilium 1.9.7)
7. Note, we have also confirmed that if you attempt `nc -zv 90.130.70.73 21` directly (this is the IP of speedtest.tele2.net), that also fails.
8. In that final test (`nc -zv 90.130.70.73 21`), we found that hubble logs a FORWARDED verdict, but tcpdump does not record any traffic to the target IP.

cc @pchaigno (worked with me on initial debugging in the community slack)

Note:
It doesn't appear to be related to this bug, but the instructions for eks tunneling setup also need correction. They recommend running this command:
helm install cilium cilium/cilium --version 1.9.7 \
  --namespace kube-system \
  --set ipam.mode=eni \
  --set egressMasqueradeInterfaces=eth0 \
  --set nodeinit.enabled=true
which results in this error:
level=info msg="Initializing IPAM" mode=eni subsys=cilium-operator-generic
level=fatal msg="eni allocator is not supported by this version of cilium-operator-generic" subsys=cilium-operator-generic

Both of the two following commands do produce valid configs:
helm install cilium cilium/cilium --version 1.9.7 \
  --namespace kube-system \
  --set ipam.mode=eni \
  --set egressMasqueradeInterfaces=eth0 \
  --set nodeinit.enabled=true

and

helm install cilium cilium/cilium --version 1.9.7 \
  --namespace kube-system \
  --set eni=true \
  --set ipam.mode=eni \
  --set egressMasqueradeInterfaces=eth0 \
  --set nodeinit.enabled=true

**Comment by thejosephstevens** [comment]
Just confirmed that the same error (connection hang) occurs when using native routing from:
helm install cilium cilium/cilium --version 1.9.7 \
  --namespace kube-system \
  --set eni=true \
  --set ipam.mode=eni \
  --set egressMasqueradeInterfaces=eth0 \
  --set nodeinit.enabled=true

**Comment by pchaigno** [comment]
> It doesn't appear to be related to this bug, but the instructions for eks tunneling setup also need correction. They recommend running this command:
helm install cilium cilium/cilium --version 1.9.7
--namespace kube-system
--set ipam.mode=eni
--set egressMasqueradeInterfaces=eth0
--set nodeinit.enabled=true
which results in this error:

Could you link to where you find those instructions in the docs? I can't seem to find the same.

**Comment by thejosephstevens** [comment]
Let me double check that, not sure if I copied it wrong or tested wrong, but going from here
https://docs.cilium.io/en/v1.9/gettingstarted/k8s-install-eks/#deploy-cilium

**Comment by thejosephstevens** [comment]
So, I'm about to try upgrading to 1.10 to see if that addresses the issue, but I am seeing this failure from the test suite. Everything else succeeds

------------------------

🔌 [pod-to-world-toFQDNs] Testing cilium-test/client2-657df6649d-zrtb2 -> google.com:80...
---------------------------------------------------------------------------------------------------------------------
❌ curl command "curl -w %{local_ip}:%{local_port} -> %{remote_ip}:%{remote_port} = %{response_code}\n --show-error --silent --fail --show-error --connect-timeout 5 --output /dev/null http://google.com" failed: command terminated with exit code 28
ℹ️  curl error: curl: (28) Resolving timed out after 5000 milliseconds

❌ [pod-to-world-toFQDNs] cilium-test/client2-657df6649d-zrtb2 (192.168.0.22) -> google.com (google.com)

**Comment by thejosephstevens** [comment]
Just confirmed the same behavior is present with Cilium 1.10

**Comment by thejosephstevens** [comment]
All right, so I learned two more things today:
First, I can see the same failure behavior while exec'd into the cilium-agent (wanted to get onto a pod with host networking).
Second, and unfortunately more relevant, I just reproduced this on a cluster with the exact same config before installing cilium (with just AWS CNI running), so I'm going to close this as not a Cilium bug and hold my breath as I descend into Amazon Linux 2 to try and figure out what's going on.

Thanks so much for helping me debug on this (shout out to @pchaigno ), I'll chime back in when I figure out what was going on (in case anyone else finds their-self in a similar rabbit hole).

**Comment by pchaigno** [comment]
> Second, and unfortunately more relevant, I just reproduced this on a cluster with the exact same config before installing cilium (with just AWS CNI running), so I'm going to close this as not a Cilium bug and hold my breath as I descend into Amazon Linux 2 to try and figure out what's going on.

Note if you want to install Cilium in chaining mode with AWS CNI, the guide to follow is https://docs.cilium.io/en/stable/gettingstarted/cni-chaining-aws-cni/ and the Helm command is fairly different. Otherwise, you should remove the `aws-node` daemonset as per the ENI guide: https://docs.cilium.io/en/stable/gettingstarted/k8s-install-helm/
```
kubectl -n kube-system delete daemonset aws-node
```

Also, in AWS-CNI chaining mode, version 1.7.9 of the AWS VPC CNI plugin is required for Cilium. I didn't realized before you wanted to run with AWS CNI.

**Comment by thejosephstevens** [comment]
Sorry, to clarify, I'm not interested in running in chaining mode with the aws cni (vxlan tunneling has worked great for us, we run too many pods per node to give an IP per pod). The test that I ran last night showed that the failing behavior occurs before the normal bootstrap steps that I run to set up Cilium (removing aws cni, installing cilium, and rolling nodes), which shows that this bugged behavior is unrelated to Cilium.

**Comment by thejosephstevens** [comment]
Just to close the loop here, apparently there are long-standing issues with connecting to FTP servers from clients in AWS (such as [this](https://stackoverflow.com/questions/27160810/ftp-access-from-aws-vpc-private-subnet-behind-a-nat-instance)). If anyone else does run into this, we were able to get around the problem by using SFTP instead of FTP or FTPS. It appears likely that in order to get FTP or FTPS working there would be changes required in instance-level Security Groups and Subnet Network ACLs.

Also to confirm, the connection failures that we did see appear to have nothing to do with Cilium.
